src/prompts/ddl/ddl-banner.tsx

```diff
@@ -0,0 +1,114 @@
+import { Component, h, createRef, RefObject } from 'preact';
+import { DDLBannerPromptConfig } from '../../core';
+import DeeplinkButton from './deeplink-button';
+
+const styles = {
+    bannerIconStyle: {
+        backgroundRepeat: 'no-repeat',
+        backgroundPosition: 'center',
+        backgroundSize: 'cover'
+    },
+    closeIconStyle: {
+        width: 12,
+        height: 12
+    }
+};
+
+export interface DDLBannerProps {
+    config: DDLBannerPromptConfig;
+    onConfirm: (config: DDLBannerPromptConfig) => void;
+    onCancel: (config: DDLBannerPromptConfig) => void;
+    dimensions: (width: number, height: number) => void;
+}
+
+export class DDLBanner extends Component<DDLBannerProps, never> {
+    private containerRef: RefObject<HTMLDivElement>;
+
+    constructor(props: DDLBannerProps) {
+        super(props);
+
+        this.containerRef = createRef<HTMLDivElement>();
+    }
+
+    componentDidMount() {
+        if (!this.containerRef.current) {
+            return;
+        }
+
+        const { clientWidth, clientHeight } = this.containerRef.current;
+
+        this.props.dimensions(clientWidth, clientHeight);
+    }
+
+    onConfirm = () => {
+        this.props.onConfirm(this.props.config);
+    };
+
+    onCancel = () => {
+        this.props.onCancel(this.props.config);
+    };
+
+    render() {
+        const { config } = this.props;
+
+        const { canonicalLinkUrl, position, labels, colors, imageUrl } = config;
+        const { heading, body, acceptAction } = labels.bannerdl;
```

Comment:
Naming, `bannerdl` prop a little odd/feels off, can we have something better / will it be refactored?

Reply:
Yip, not 100% with it myself - quick naming there, got a note about it here too! names in a hat! (started as 'app_acquisition_banner', then 'deep_link_banner', full of underscores 😂 ) - still not satisfied came up with a good moniker for it yet tho!

Reply:
Ahh - its was just reusing the banner config exactly hence the nested naming - but agree, can clean this up and remove it which I've done now 👍